### PR TITLE
add test for cudf send receive

### PR DESCRIPTION
**Pull request by quasiben**
PR adds a test for cudf send/receive and mimics the scheme dask uses for sending cuda objects through UCX.  

This test is currently failing and should serve as a minimal example of distributed/cudf working with UCX

cc @mrocklin @Akshay-Venkatesh @TomAugspurger @rjzamora

### CODE REVIEW

File: tests/test_send_recv_obj.py [hunk 1]
```diff
@@ -172,3 +173,155 @@
...
+    size = len(cdf)
+
+    cuda_info = {'shape': [3,], 'typestr': '<i8'}
+
+    async with echo_pair(cuda_info) as (_, client):
+        await client.send_obj(bytes(str(size), encoding='utf-8'))
+        await client.send_obj(cudf_frame)
+        resp = await client.recv_obj(size, cuda=True)
+        result = ucp.get_obj_from_msg(resp)
+
```
Comment (quasiben) [inline tests/test_send_recv_obj.py]: @Akshay-Venkatesh this test currently fails -- after converting the buffer back to a cupy/cudf object the array is [7, 0, 0] when it should be [1, 2, 3]

Comment (Akshay-Venkatesh) [reply tests/test_send_recv_obj.py]: Ben, this is a basic question but I'm not able to get cudf running. I tried the following:
```
python3 -m pip install cudf
```
But import fails with the following error:
```
akvenkatesh@prm-dgx-11:~/ucx-py$ python3
Python 3.7.2 (default, Mar  4 2019, 21:03:35) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cudf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/akvenkatesh/py/install/lib/python3.7/site-packages/cudf/__init__.py", line 3, in <module>
    from cudf import dataframe
  File "/home/akvenkatesh/py/install/lib/python3.7/site-packages/cudf/dataframe/__init__.py", line 1, in <module>
    from cudf.dataframe import (buffer, dataframe, series,
  File "/home/akvenkatesh/py/install/lib/python3.7/site-packages/cudf/dataframe/dataframe.py", line 24, in <module>
    from cudf import formatting, _gdf
  File "/home/akvenkatesh/py/install/lib/python3.7/site-packages/cudf/_gdf.py", line 15, in <module>
    import nvcategory
  File "/home/akvenkatesh/py/install/lib/python3.7/site-packages/nvcategory.py", line 2, in <module>
    import pyniNVCategory
ImportError: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
```

Should I try building from source?

Comment (quasiben) [reply tests/test_send_recv_obj.py]: Sorry I missed this -- we chatted briefly about installing from source and a small reproducible would be just using rmm:

```python
from librmm_cffi import librmm as rmm
import numpy as np
devicearray = rmm.to_device(np.array[1, 2, 3]))
```

**Comment by quasiben** [comment]
Currently failing RMM send/receive with this error:

> numba.cuda.cudadrv.driver.CudaAPIError: [1] Call to cuMemcpyDtoH results in CUDA_ERROR_INVALID_VALUE

**Comment by quasiben** [comment]
Note RMM *should* be equivalent the numba tests

**Comment by quasiben** [comment]
The cudf test is partially working now!  Thanks to @Akshay-Venkatesh and @pentschev!!

The issues largely stem from many small issues:
- not setting shape/typestr in the returned buffer manually
- distributed nbytes function with numba < 0.45 returns the length of the numba device array instead of the size (https://github.com/numba/numba/pull/3933)  -- @TomAugspurger is amazing!

I also need to look at few other items @Akshay-Venkatesh mentioned late on Friday

File: tests/test_send_recv_obj.py [hunk 2]
```diff
@@ -172,3 +178,225 @@
...
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "g",
+    [
+        lambda cudf: cudf.Series([1, 2, 3]),
+        lambda cudf: cudf.Series([1, 2, 3], index=[4, 5, 6]),
+        lambda cudf: cudf.Series([1, None, 3]),
+        lambda cudf: cudf.DataFrame({'a': range(2**26)}),
```
Comment (quasiben) [inline tests/test_send_recv_obj.py]: Large dataframe testing

**Comment by mrocklin** [comment]
What is the status here?  Should this be merged?

**Comment by quasiben** [comment]
This can closed

**Pull request closed without merging**